From: Thomas Klute <thomas2.klute@uni-dortmund.de>
Date: Wed, 8 Feb 2017 13:27:17 +0100
Subject: Check if flock supports --verbose

Some old versions of flock do not support the --verbose option, namely
the one in Debian Jessie. Check for support at configure time and
enable the option only if available.
---
 configure.ac            | 12 +++++++++++-
 test/proxy_backend.bash |  2 +-
 test/runtests           |  2 +-
 3 files changed, 13 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index 0577435..425f2b8 100644
--- a/configure.ac
+++ b/configure.ac
@@ -83,8 +83,18 @@ AS_IF([test "${FLOCK}" != "no"],
 	lockfile="$(mktemp)"
 	AS_IF([${FLOCK} --timeout 1 ${lockfile} true >&AS_MESSAGE_LOG_FD 2>&1],
 	      [flock_works="yes"], [flock_works="no"])
-	rm "${lockfile}"
 	AC_MSG_RESULT([$flock_works])
+	# Old versions of flock do not support --verbose. They fail
+	# without executing the command but still return 0. Check for
+	# this behavior by testing if the rm command was executed.
+	AC_MSG_CHECKING([whether ${FLOCK} supports --verbose])
+	testfile="$(mktemp)"
+	AS_IF([${FLOCK} --verbose --timeout 1 ${lockfile} rm "${testfile}" \
+			>&AS_MESSAGE_LOG_FD 2>&1; test ! -e "${testfile}"],
+	      [flock_verbose="yes"; FLOCK="${FLOCK} --verbose"],
+	      [flock_verbose="no"; rm "${testfile}"])
+	AC_MSG_RESULT([$flock_verbose])
+	rm "${lockfile}"
       ],
       [flock_works="no"])
 # disable flock if requested by user or it doesn't support timeout
diff --git a/test/proxy_backend.bash b/test/proxy_backend.bash
index 5b1bafe..c2d8507 100644
--- a/test/proxy_backend.bash
+++ b/test/proxy_backend.bash
@@ -37,7 +37,7 @@ function backend_apache
 			 "locking."
 		    flock_cmd=""
 		elif [ -n "${lockfile}" ]; then
-		    flock_cmd="${FLOCK} --verbose -w ${TEST_LOCK_WAIT} ${lockfile}"
+		    flock_cmd="${FLOCK} -w ${TEST_LOCK_WAIT} ${lockfile}"
 		else
 		    echo "Locking disabled, using wait based on proxy PID file."
 		    wait_pid_gone "${BACKEND_PID}"
diff --git a/test/runtests b/test/runtests
index 718b27f..c1ff135 100755
--- a/test/runtests
+++ b/test/runtests
@@ -159,7 +159,7 @@ if [ -n "${USE_TEST_NAMESPACE}" ]; then
     echo "Using namespaces to isolate tests, no need for locking."
     flock_cmd=""
 elif [ -n "${TEST_LOCK}" ]; then
-    flock_cmd="${FLOCK} --verbose -w ${TEST_LOCK_WAIT} $(realpath ${TEST_LOCK})"
+    flock_cmd="${FLOCK} -w ${TEST_LOCK_WAIT} $(realpath ${TEST_LOCK})"
 else
     echo "Locking disabled, using wait based on Apache PID file."
     wait_pid_gone "${TEST_PID}"
